reconcile/utils/saasherder.py

```diff
@@ -859,6 +859,91 @@ def update_moving_commit(self, job_spec):
             f"{namespace_name}/{env_name}/{ref}"
         self.state.add(key, value=commit_sha, force=True)
 
+    def get_upstream_jobs_diff(self, dry_run, current_state):
+        results = threaded.run(self.get_upstream_jobs_diff_saas_file,
+                               self.saas_files,
+                               self.thread_pool_size,
+                               dry_run=dry_run,
+                               current_state=current_state)
+        return [item for sublist in results for item in sublist]
```

Comment:
I find `itertools.chain` easier. According to stackoverflow it's also faster.

```python
return itertools.chain.from_iterable(results)
```
(it returns a generator, wrap it onto a `list` if you truly want to have a list)

Reply:
But this is just a nit.

Reply:
this is not bad. there are a lot of places in the code base that can use this change!

Reply:
https://github.com/app-sre/qontract-reconcile/pull/1751